### PR TITLE
Add support for Module.dynamicLibraries in pthreads

### PR DESCRIPTION
**Pull request by jprendes**
Currently `Module.dynamicLibraries` is only accounted for in the main module.
This change makes pthreads aware of this property.

Add the value of `Module.dynamicLibraries` to the `'load'`'s `postMessage` to the pthread worker.

### CODE REVIEW

**Comment by sbc100** [comment]
Thanks!

So the build time setting RUNTIME_LINKED_LIBS was being honored by the runtime dynamicLibraries what not?

Does this work as is, can you add test?

**Comment by jprendes** [comment]
Yes, exactly. `RUNTIME_LINKED_LIBS` is hardcoded in the Module at compile time, so each worker gets a copy of it when loading the module.
When `Module.dynamicLibraries` is set at runtime (e.g. in the HTML) the workers don't automatically get a copy.
I am updating the PR to include a test.

File: tests/test_other.py [hunk 1]
```diff
@@ -1465,6 +1465,49 @@ def test_dynamic_link_with_exceptions_and_assetions(self):
...
+    create_test_file('test.js', '''
+      var createMyModule = require('./main.js');
+      createMyModule({
+        dynamicLibraries: ['side.wasm']
+      });
+    ''')
+
+    self.node_args += ['--experimental-wasm-threads', '--experimental-wasm-bulk-memory']
+    seen = self.run_js('test.js', assert_returncode=0)
+    self.assertContained(['42'], seen)
```
Comment (sbc100) [inline tests/test_other.py:1509]: Can you move this alongside the existing `test_pthread_dynamic_linking` test in test_core.py?     You should be able to use the `dylink_test` helper (with `auto_load=False`) to simplify the test.

File: tests/test_core.py [hunk 2]
```diff
@@ -8229,6 +8229,42 @@ def test_pthread_dynamic_linking(self):
...
+    self.emcc_args += ['-pthread', '-Wno-experimental']
+    self.set_setting('PROXY_TO_PTHREAD')
+    self.set_setting('EXIT_RUNTIME')
+    self.set_setting('MODULARIZE')
+    self.set_setting('EXPORT_NAME', 'createMyModule')
+
+    def post(filename):
+      shutil.move(filename, '{filename}.module.js')
+      create_test_file(filename, '''
+        var createMyModule = require('./{filename}.module.js');
```
Comment (jprendes) [inline tests/test_core.py:8246]: I am not completely sure how robust this is.
It assumes that filename is a relative path and that the test runs in node (or an environment with `require`).
Seems to work fine locally.

Comment (sbc100) [reply tests/test_core.py:8246]: Is "{filename}" supposed to be a sting substitution in python here?   If you use would need to use the `f` prefix on the string..

Comment (jprendes) [reply tests/test_core.py:8246]: Yes, thanks!
Python is not my main (or non-main) language

Comment (sbc100) [reply tests/test_core.py:8246]: I think you could alternatively using `--extern-pre-js` and `--extern-post-js` which might be cleaner.

Comment (jprendes) [reply tests/test_core.py:8246]: Great, that sounds much neater.

**Review by sbc100 — APPROVED** [review]

**Pull request merged**